### PR TITLE
New MarkUs Nav Bar Dropdown Menu

### PR DESCRIPTION
**Pull request by therealfake**
## Motivation and Context

The current navigation bar at the very top has alot of information such as user's name, user's username, setting button, and a log out button. The motivation for combining this information and the information in the footer ("About MarkUs" and "GitHub issues list") into a dropdown menu is to make the interface look more organized and provide seamless navigation.
image

## Your Changes

**Description**:
- Moved user's information and buttons in the top right corner of webpage into a dropdown menu. 
- Moved links in footers to this dropdown menu
- Removed footer and associated styling
- Fixed mobile view navigation bar logo styling

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

First tested locally, for which all MarkUs test suite tests passed.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have updated the Changelog.md file.

### CODE REVIEW

File: app/assets/stylesheets/common/_markus.scss [hunk 1]
```diff
@@ -1004,6 +1004,7 @@ nav {
     position: absolute;
     text-align: left;
     top: 1.95em;
+    width: 101%;
```
Comment (david-yz-liu) [inline app/assets/stylesheets/common/_markus.scss:1007]: So rather than do this, I think it's okay to change the `min-width` setting to be the same as the parent's. (I believe changing that will also allow the dropdown to be *wider* than the parent if the text is longer, though to be honest I'm not sure. Perhaps you could try it?)

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -5,40 +5,58 @@
...
+                            clear_role_switch_session_course_path(@current_course) %>
+              </li>
+            <% end %>
+            <li>
+              <%= link_to t(:about_markus),
+                          about_main_index_path,
+                          remote: true %>
+            </li>
+            <li>
+            <span class="issues">
```
Comment (david-yz-liu) [inline app/views/layouts/_header.html.erb:46]: I think you can remove the "span" part and the class, and just keep the `a` tag.

**Review by david-yz-liu — COMMENTED** [review]
@therealfake thank you! I did a preliminary review and left a couple of small comments. Also, here are two bigger things I'd like you to address/consider:

1. You'll need to handle the different layout on mobile/smaller screens. Try using your browser's developer tools to switch screen size to a mobile screen and seeing what your change looks like. If you need to create different style rules for module, you can do it using `breakpoint(mobile)`, as we do in the existing CSS.
2. I'd like to keep the icons we used for the existing buttons, and also add new icons for the "about" and "Github issues list" links. (Maybe use the "info i" and GitHub logos, respectively.) Try playing around with the icon and text alignments to see what looks good. E.g., icons and text left-aligned, icons left-aligned and text right-aligned, etc.

File: app/views/layouts/_header.html.erb (hunk 2 above)
Comment (therealfake) [reply app/views/layouts/_header.html.erb:46]: Done

File: app/assets/stylesheets/common/_markus.scss (hunk 1 above)
Comment (therealfake) [reply app/assets/stylesheets/common/_markus.scss:1007]: I found that for both, if a single word is short than about 33 characters the new words will be on the next line if they cannot fit on the existing line. When this is not the case for a single word, min-width does make the dropdown box larger to fit and width makes the box the same size and the test goes outside to the right of it.

Comment (david-yz-liu) [reply app/assets/stylesheets/common/_markus.scss:1007]: @therealfake I'm having trouble picturing what you're describing. Can you attach screenshots with this rule toggled on/off to illustrate what you mean?

Comment (therealfake) [reply app/assets/stylesheets/common/_markus.scss:1007]: @david-yz-liu Sure!
So this image shows when each word itself is < ~25 chars it wraps to the next line and the box is unaffected:
image
This is what happens when we just go with ```width: 101%```:
(The word is "AboutMarkUsAboutMarkUsAboutMarkUs" which is 33 chars)
image
And this is what happens when we just go with ```min-width: 101%```:
image

File: app/assets/stylesheets/common/_navigation.scss [hunk 3]
```diff
@@ -454,7 +391,6 @@ li#dropdown {
   ul {
     li {
       border-radius: 0;
-
```
Comment (david-yz-liu) [inline app/assets/stylesheets/common/_navigation.scss:457]: Revert this change

File: app/assets/stylesheets/common/_navigation.scss [hunk 4]
```diff
@@ -419,10 +359,7 @@ li#dropdown {
 
     ul {
       background: $background-main;
-      border-left: 0;
-      border-radius: 0;
-      border-right: 0;
-      border-top: 0;
+      border: 0;
```
Comment (david-yz-liu) [inline app/assets/stylesheets/common/_navigation.scss:362]: Revert this change

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -5,40 +5,60 @@
     <% end %>
   </div>
   <div id='user-info'>
-    <span id='logout_area'>
-      <span id='instructor_logged_in_as'>
+    <ul>
+      <li id="user-info-dropdown">
+        <div class = "dropdown">
```
Comment (david-yz-liu) [inline app/views/layouts/_header.html.erb:10]: remove the spaces around `=`

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -5,40 +5,60 @@
...
+        <span id='instructor_logged_in_as'>
         <% if allowed_to?(:role_is_switched?) %>
           <strong><%= @real_user.display_name %></strong>
```
Comment (david-yz-liu) [inline app/views/layouts/_header.html.erb:13]: Let's put this name in parentheses (previously, the `@current_user` as in parentheses), to maintain a split between them.

You should test out the UI for the dropdown after switching roles to see this. *ALSO*, please add an icon for the "Cancel role switch" action. (Use the same icon that currently appears on master.)

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -5,40 +5,60 @@
...
-                  id: 'logout_link',
-                  method: :post,
-                  class: 'button inline-button') %>
-    </span>
+          <ul>
+            <% unless allowed_to?(:role_is_switched?) %>
+              <li>
+                <%= link_to t('menu.settings'),
+                            settings_users_path,
+                            class: 'settings'%>
```
Comment (david-yz-liu) [inline app/views/layouts/_header.html.erb:25]: Add a space before `%>` (occurs repeated below, as well)

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -5,40 +5,60 @@
...
+              </li>
+            <% end %>
+            <li>
+              <%= link_to t(:about_markus),
+                          about_main_index_path,
+                          remote: true,
+                          class: 'about-markus'%>
+            </li>
+            <li>
+              <a class = 'github' href="https://github.com/MarkUsProject/Markus/issues/"><%= t(:issues) %></a>
```
Comment (david-yz-liu) [inline app/views/layouts/_header.html.erb:49]: Remove spaces around `=`

File: app/views/layouts/content.html.erb [hunk 9]
```diff
@@ -36,8 +35,6 @@
     </section>
   </div>
```
Comment (david-yz-liu) [inline app/views/layouts/content.html.erb:37]: You can delete this line as well

File: app/assets/stylesheets/common/_markus.scss [hunk 10]
```diff
@@ -1003,11 +1003,12 @@ nav {
     display: none;
     left: -1px;
     margin: 0;
-    min-width: 16ch;
     padding: 0;
     position: absolute;
     text-align: left;
-    top: 1.95em;
+    top: 2.2em;
```
Comment (david-yz-liu) [inline app/assets/stylesheets/common/_markus.scss:1009]: I'm not sure why you changed this one?

File: app/assets/stylesheets/common/_markus.scss [hunk 11]
```diff
@@ -1003,11 +1003,12 @@ nav {
...
+    width: 101%;
```
Comment (david-yz-liu) [inline app/assets/stylesheets/common/_markus.scss:1010]: See comment below about `width: 101%`. Also the word-wrap rule you added was good! Should solve the issue you described.

File: app/views/layouts/content.html.erb (hunk 9 above)
Comment (therealfake) [reply app/views/layouts/content.html.erb:37]: Sorry I'm unsure which line you mean.

File: app/assets/stylesheets/common/_navigation.scss [hunk 12]
```diff
@@ -11,8 +11,12 @@ header#header {
   padding: 0.25em $dimen-horizontal-nav;
   width: 100%;
 
+  @include breakpoint(mobile) {
+    border-bottom: 0;
+  }
+
   #course,
-  #user_info {
+  #user-info {
```
[the remark this answers is not in the document]
Comment (therealfake) [reply app/assets/stylesheets/common/_navigation.scss:19]: Lines 30-32 can be removed without any effects. But removing line 28 will cause the dropdown bar to lose its positioning in the top right corner.

File: app/assets/stylesheets/common/_markus.scss (hunk 11 above)
Comment (therealfake) [reply app/assets/stylesheets/common/_markus.scss:1010]: 1.95em slightly overlays the display name box

File: app/assets/stylesheets/common/_markus.scss (hunk 10 above)
Comment (therealfake) [reply app/assets/stylesheets/common/_markus.scss:1009]: 1.95em slightly overlays the display name box

File: app/assets/stylesheets/common/_markus.scss (hunk 11 above)
Comment (therealfake) [reply app/assets/stylesheets/common/_markus.scss:1010]: Declaring with a variable for 200px works for the normal view, however once switched into a role, the dropdown becomes much smaller than the box, even if I switch to using min-width.

File: app/assets/stylesheets/common/_navigation.scss (hunk 12 above)
[the remark this answers is not in the document]
Comment (david-yz-liu) [reply app/assets/stylesheets/common/_navigation.scss:19]: Okay, you can remove just Lines 30-32 then.

File: app/assets/stylesheets/common/_navigation.scss [hunk 13]
```diff
@@ -358,9 +285,10 @@ footer {
     border: 0;
     text-align: left !important;
   }
-
-  #course,
-  #user_info {
+  //
```
[the remark this answers is not in the document]
Comment (david-yz-liu) [reply app/assets/stylesheets/common/_navigation.scss:288]: You may have missed this.

File: app/assets/stylesheets/common/_markus.scss (hunk 10 above)
Comment (david-yz-liu) [reply app/assets/stylesheets/common/_markus.scss:1009]: @therealfake it looks fine to me. Please revert.

File: app/assets/stylesheets/common/_navigation.scss (hunk 12 above)
[the remark this answers is not in the document]
Comment (david-yz-liu) [reply app/assets/stylesheets/common/_navigation.scss:19]: @therealfake okay, you can just remove lines 30-32 👍
